### ISSUE TITLE
Transcribe result extraction too time consuming

### ISSUE THREAD

**Issue by dustinjoe** [guillaumekln/faster-whisper#117]
Hi, sorry for interrupting. And thanks for this great work!
I am trying to combine this faster whisper with Pyannote speaker diarization following the code here:
https://github.com/yinruiqing/pyannote-whisper/blob/main/pyannote_whisper/utils.py
According to my experiment, 
It took  0.5442 second(s) to Whisper transcribe using this faster whisper.
It took  11.9825 second(s) to Pyannote diarize.
It took  19.0851 second(s) to merge results.
So in the merging part, most of the time is spent on the transcribe result extraction function:
It took  19.0565 second(s) to extract from whisper result.
It took  0.0284 second(s) to add spk info to text.
It took  0.0002 second(s) to merge sentences.

The transcribe extraction function (modified a little bit based on your format here) is creating a list of sentence timestamps and text contents:
 ```
def get_text_with_timestamp_fromsegments(transcribe_segments):
    timestamp_texts = []
    for item in transcribe_segments:
        start = item.start
        end = item.end
        text = item.text
        timestamp_texts.append((Segment(start, end), text))
    return timestamp_texts
 ```
So basically, this function takes most of the execution time. Wondering if you can give some suggestions on how to extract transcribe results faster here? Thank you!

**Comment by guillaumekln** [comment]
Hi,

See issue #67.

Basically the model returns a **generator** over segments. So the transcription only starts when you iterate over the segments.

If you don't like this behavior, you can get all segments in a list:

```python
segments, _ = model.transcribe("audio.mp3")
segments = list(segments)  # The transcription will run in this line.
```

**Comment by dustinjoe** [comment]
Oh, thanks for your fast reply!
Sorry for my misunderstanding.
